projects/graph/graph.py

```diff
@@ -7,42 +7,112 @@ class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
         self.vertices = {}
+        self.queue = Queue()
+        self.stack = Stack()
     def add_vertex(self, vertex):
         """
         Add a vertex to the graph.
         """
-        pass  # TODO
+        self.vertices[vertex] = set()
     def add_edge(self, v1, v2):
         """
         Add a directed edge to the graph.
         """
-        pass  # TODO
+        if v1 not in self.vertices:
```

Comment:
Good clean code Lorenzo. If `v1` or our `origin vertex` doesn't exist add it to the graph. Otherwise, add  `v2` our `destination vertex` to `v1` vertices/edges 👍